logstash-core/lib/logstash/api/commands/stats.rb

```diff
@@ -45,14 +60,22 @@ def events
           )
         end
 
-        def pipeline(pipeline_id = nil)
+        def pipeline(pipeline_id = nil, opts={})
           if pipeline_id.nil?
             pipeline_ids = service.get_shallow(:stats, :pipelines).keys
+            extended_stats = LogStash::Config::PipelinesInfo.format_pipelines_info(
```

Comment:
since we need the `extended_stats` either way, let's bring this up above the `if` clause so we can avoid repetition.

Also, as I [previously suggested](https://github.com/elastic/logstash/pull/10576#pullrequestreview-248312786), indexing the extended stats into a `Hash{String=>Hash}` will incur a small cost once that will drastically reduce the repetitive cost of lookups.

Reply:
I think I've resolved this concern now. I did repeat the logic a bit so there is some room for DRY improvements but I thought this might be more readable. Happy to change it though if you think it would be cleaner another way. :)